src/main.cpp

```diff
@@ -4214,8 +4243,16 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
     // Check transactions
     bool fZerocoinActive = block.GetBlockTime() > Params().Zerocoin_StartTime();
     vector<CBigNum> vBlockSerials;
+    // TODO: Check if this is ok... blockHeight is always the tip or should we look for the prevHash and get the height?
```

Comment:
It definitely does seem like it would be better to take context (block height) out of `CheckTransaction()` and into one of the contextual functions. Probably a goal to work towards, as it would take some good refactoring to do it.